api/src/main/kotlin/org/jellyfin/apiclient/api/client/KtorClient.kt

```diff
@@ -0,0 +1,141 @@
+package org.jellyfin.apiclient.api.client
+
+import com.google.gson.FieldNamingPolicy
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import org.jellyfin.apiclient.api.client.adapter.LocalDateTimeTypeAdapter
+import org.jellyfin.apiclient.api.client.adapter.UUIDTypeAdapter
+import java.time.LocalDateTime
+import java.util.*
+
+open class KtorClient(
+	var baseUrl: String
+) {
+	val client = HttpClient {
+		install(JsonFeature) {
+			serializer = GsonSerializer {
+				setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)
+				serializeNulls()
+				registerTypeAdapter(UUID::class.java, UUIDTypeAdapter())
+				registerTypeAdapter(LocalDateTime::class.java, LocalDateTimeTypeAdapter())
+			}
```

Comment:
I've recently read Gson shouldn't be used anymore, maybe we should switch to something more modern and Kotlin-aware like kotlinx.serialization for the new client?

Reply:
kotlinx.serialization requires an annotation AFAIK but the `model` module doesn't have external dependencies and I want to keep it that way.
Do you have a source about not using gson, I'm quite intested in that.

Reply:
Yep, kotlinx.serialization does require an annotation unfortunately. [Here's a comment](https://www.reddit.com/r/Kotlin/comments/exmp2s/json_to_kotlin_data_class/fgak5mc/) from Jake Wharton on the state of Gson from already a few months ago. It was recently references in [this](https://www.reddit.com/r/Kotlin/comments/ikwecd/whats_the_best_json_to_kotlin_objects_converter/) thread.

Reply:
Ah thanks, I might switch to kotlinx.serialization in a future PR. I'd love to make the apiclient a multiplatform project but that will be quite some work because simple things like UUID's are not in Kotlin's stdlib and there is no date/time implementation yet.